### PR TITLE
git_retry_tag and fixes.

### PR DESCRIPTION
**Pull request by nbartos**

### CODE REVIEW

File: functions.sh [hunk 1]
```diff
@@ -103,6 +103,41 @@ git_retry_fetch() {
     )
 }
 
+git_retry_tag() {
+    if test $# -ne 2; then
+        die "git_fetch_parent <owner> <tag>" || return 1
```
Comment (novas0x2a) [inline functions.sh]: message

File: functions.sh [hunk 2]
```diff
@@ -103,6 +103,41 @@ git_retry_fetch() {
...
+    fi
+
+    git tag "$2"
```
Comment (novas0x2a) [inline functions.sh]: this creates a shallow tag

**Comment by nbartos** [comment]
Should be good now.

**Comment by fcforrest** [comment]
LGTM :police_car:

File: functions.sh [hunk 3]
```diff
@@ -103,6 +103,41 @@ git_retry_fetch() {
     )
 }
 
+git_retry_tag() {
+    if test $# -ne 2; then
+        die "git_retry_tag <owner> <tag>" || return 1
+    fi
+
+    git tag "$2" -a -m "$2"
```
Comment (novas0x2a) [inline functions.sh]: what type of tag are you using this for? usually the message should be the same (or sameish, like including date) for any tag of the same type.

Comment (nbartos) [inline functions.sh]: It's being called by the new jenkins job to tag the new version.  Right now they are only v$RELEASE_VERSION.

Comment (novas0x2a) [inline functions.sh]: the message should be "version" for those to match the existing tags, those messages are the only way to tell tag types apart besides regexes

Comment (nbartos) [inline functions.sh]: I changed it to pass in another arg for the message, then changed the jenkins job to pass in "version $RELEASE_VERSION" for that arg.

**Pull request merged**